data/meterpreter/meterpreter.py

```diff
@@ -566,6 +569,36 @@ def handle_dead_resource_channel(self, channel_id):
 		pkt  = struct.pack('>I', len(pkt) + 4) + pkt
 		self.send_packet(pkt)
 
+	def _core_machine_id(self, request, response):
+		serial = ''
+		machine_name = platform.uname()[1]
+		if has_windll:
+			from ctypes import wintypes
+
+			k32 = ctypes.windll.kernel32
+			sys_dir = ctypes.create_unicode_buffer(260)
+			if not k32.GetSystemDirectoryW(ctypes.byref(sys_dir), 260):
+				return ERROR_FAILURE
```

Comment:
Returning ```ERROR_FAILURE_WINDOWS``` here will cause the error result returned to metasploit to include the value of ```kernel32.GetLastError()``` which might be helpful.